tracer/src/Datadog.Trace/AppSec/Waf/Waf.cs

```diff
@@ -91,6 +91,40 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
 
+        public bool UpdateRules(IEnumerable<RuleData[]> res)
+        {
+            var finalRuleDatas = MergeRuleDatas(res);
+
+            var encoded = encoder.Encode(finalRuleDatas, new List<Obj>(), false);
+
+            var ret = wafNative.UpdateRuleData(ruleHandle.Value, encoded.RawPtr);
+            return ret == DDWAF_RET_CODE.DDWAF_OK;
+        }
+
+        internal static List<object> MergeRuleDatas(IEnumerable<RuleData[]> res)
```

Comment:
here we receive several files of content
```
 "rules_data": [
        …,
        {
            "id": “blocklist”,
            "type": “data_with_expiration”,
            "data": [
                 …,
                 { “value”: “127.0.0.1”, “expiration”: … },
                 …,
            ]
        },
        …
    ]

```
 As a result, multiple Rule Data may use the same DATA_ID and DATA_TYPE. In this case, all values must be merged
When a value is associated with an expiration date, the [latest date takes precedence](https://docs.google.com/document/d/1j314dxWmTwXfncFWvNERQdjoscXWLnKuype0VfzMRfY/edit#heading=h.mmag5q2w66nd)


For the sake of readability and convenience, I used linq here, I believe without linq it would be very long and cumbersome 😅